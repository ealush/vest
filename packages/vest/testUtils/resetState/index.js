import { OPERATION_MODE_STATEFUL } from '../../src/constants';
import * as state from '../../src/core/state';
import getState from '../../src/core/suite/getState';
import runRegisterSuite from '../runRegisterSuite';

const resetState = (suiteId, suiteName = suiteId) => {
  state.set(() => null);
  state.register();

  if (suiteName) {
    runRegisterSuite({
      suiteId,
      name: suiteName,
      operationMode: OPERATION_MODE_STATEFUL,
    });
    return getState(suiteId);
  }
};

export default resetState;
