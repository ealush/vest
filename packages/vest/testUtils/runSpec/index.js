import { VEST_DIST_BUILDS } from '../vestBuilds';

const vest = require('../../src');

const runSpec = callback => {
  callback(vest);

  if (!global.isWatchMode) {
    VEST_DIST_BUILDS.forEach(callback);
  }
};

export default runSpec;
