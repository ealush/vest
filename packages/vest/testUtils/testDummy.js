/* eslint-disable jest/no-export */
import faker from 'faker';

import vest from '../src';

/**
 * Generates dummy vest tests.
 * @param {Object} [vestRef] Reference to vest build.
 */
const testDummy = (vestRef = vest) => {
  const { test } = vestRef;
  const failing = (
    name = faker.random.word(),
    statement = faker.random.words()
  ) =>
    test(
      name,
      statement,
      jest.fn(() => {
        throw new Error();
      })
    );

  const failingWarning = (
    name = faker.random.word(),
    statement = faker.random.words()
  ) =>
    test(
      name,
      statement,
      jest.fn(() => {
        vest.warn();
        throw new Error();
      })
    );

  const passing = (
    name = faker.random.word(),
    statement = faker.random.words()
  ) => test(name, statement, jest.fn());

  const passingWarning = (
    name = faker.random.word(),
    statement = faker.random.words()
  ) =>
    test(
      name,
      statement,
      jest.fn(() => {
        vest.warn();
      })
    );

  const failingAsync = (
    name = faker.random.word(),
    { statement, time = 0 } = {}
  ) =>
    test(
      name,
      statement,
      jest.fn(
        () =>
          new Promise((resolve, reject) => {
            setTimeout(reject, time);
          })
      )
    );

  const failingWarningAsync = (
    name = faker.random.word(),
    { statement, time = 0 } = {}
  ) =>
    test(
      name,
      statement,
      jest.fn(() => {
        vest.warn();
        return new Promise((resolve, reject) => {
          setTimeout(reject, time);
        });
      })
    );

  const passingAsync = (
    name = faker.random.word(),
    { statement, time = 0 } = {}
  ) =>
    test(
      name,
      statement,
      jest.fn(
        () =>
          new Promise(resolve => {
            setTimeout(resolve, time);
          })
      )
    );

  const passingWarningAsync = (
    name = faker.random.word(),
    { statement, time = 0 } = {}
  ) =>
    test(
      name,
      statement,
      jest.fn(() => {
        vest.warn();
        return new Promise(resolve => {
          setTimeout(resolve, time);
        });
      })
    );

  return {
    failing,
    failingAsync,
    failingWarning,
    failingWarningAsync,
    passing,
    passingAsync,
    passingWarning,
    passingWarningAsync,
  };
};

export default testDummy;

export const dummyTest = testDummy();
