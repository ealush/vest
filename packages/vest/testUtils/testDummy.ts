/* eslint-disable jest/valid-title */
import faker from 'faker';

import { test, warn } from 'vest';

/**
 * Generates dummy vest tests.
 */
const testDummy = () => {
  const failing = (
    name: string = faker.random.word(),
    message: string = faker.random.words(),
    groupName?: string
  ) => {
    const to = test(
      name,
      message,
      jest.fn(() => {
        throw new Error();
      })
    );

    if (groupName) {
      to.groupName = groupName;
    }

    return to;
  };

  const failingWarning = (
    name = faker.random.word(),
    message = faker.random.words(),
    groupName?: string
  ) => {
    const to = test(
      name,
      message,
      jest.fn(() => {
        warn();
        throw new Error();
      })
    );

    if (groupName) {
      to.groupName = groupName;
    }

    return to;
  };

  const passing = (
    name = faker.random.word(),
    message = faker.random.words(),
    groupName?: string
  ) => {
    const to = test(name, message, jest.fn());

    if (groupName) {
      to.groupName = groupName;
    }

    return to;
  };

  const passingWarning = (
    name = faker.random.word(),
    message = faker.random.words(),
    groupName?: string
  ) => {
    const to = test(
      name,
      message,
      jest.fn(() => {
        warn();
      })
    );
    if (groupName) {
      to.groupName = groupName;
    }
    return to;
  };

  const failingAsync = (
    name = faker.random.word(),
    { message = faker.random.words(), time = 0 } = {}
  ) =>
    test(
      name,
      message,
      jest.fn(
        () =>
          new Promise((_, reject) => {
            setTimeout(reject, time);
          })
      )
    );

  const failingWarningAsync = (
    name = faker.random.word(),
    { message = faker.random.words(), time = 0 } = {}
  ) =>
    test(
      name,
      message,
      jest.fn(() => {
        warn();
        return new Promise((_, reject) => {
          setTimeout(reject, time);
        });
      })
    );

  const passingAsync = (
    name = faker.random.word(),
    { message = faker.random.words(), time = 0 } = {}
  ) =>
    test(
      name,
      message,
      jest.fn(
        () =>
          new Promise(resolve => {
            setTimeout(resolve, time);
          })
      )
    );

  const passingWarningAsync = (
    name = faker.random.word(),
    { message = faker.random.words(), time = 0 } = {}
  ) =>
    test(
      name,
      message,
      jest.fn(() => {
        warn();
        return new Promise(resolve => {
          setTimeout(resolve, time);
        });
      })
    );

  return {
    failing,
    failingAsync,
    failingWarning,
    failingWarningAsync,
    passing,
    passingAsync,
    passingWarning,
    passingWarningAsync,
  };
};

export const dummyTest = testDummy();
