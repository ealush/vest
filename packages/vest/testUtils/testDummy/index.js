/* eslint-disable jest/no-export */
import faker from 'faker';
import vest from '../../src';

const testDummy = (vestRef = vest) => {
  const { test } = vestRef;
  const failing = (
    name = faker.random.word(),
    statement = faker.random.words()
  ) =>
    test(name, statement, () => {
      throw new Error();
    });

  const failingWarning = (
    name = faker.random.word(),
    statement = faker.random.words()
  ) =>
    test(name, statement, () => {
      vest.warn();
      throw new Error();
    });

  const passing = (
    name = faker.random.word(),
    statement = faker.random.words()
  ) => test(name, statement, Function.prototype);

  const passingWarning = (
    name = faker.random.word(),
    statement = faker.random.words()
  ) =>
    test(name, statement, () => {
      vest.warn();
    });

  const failingAsync = (
    name = faker.random.word(),
    { statement, time = 0 } = {}
  ) =>
    test(
      name,
      statement,
      () =>
        new Promise((resolve, reject) => {
          setTimeout(reject, time);
        })
    );

  const failingWarningAsync = (
    name = faker.random.word(),
    { statement, time = 0 } = {}
  ) =>
    test(name, statement, () => {
      vest.warn();
      return new Promise((resolve, reject) => {
        setTimeout(reject, time);
      });
    });

  const passingAsync = (
    name = faker.random.word(),
    { statement, time = 0 } = {}
  ) =>
    test(
      name,
      statement,
      () =>
        new Promise(resolve => {
          setTimeout(resolve, time);
        })
    );

  const passingWarningAsync = (
    name = faker.random.word(),
    { statement, time = 0 } = {}
  ) =>
    test(name, statement, () => {
      vest.warn();
      return new Promise(resolve => {
        setTimeout(resolve, time);
      });
    });

  return {
    failing,
    passing,
    failingAsync,
    passingAsync,
    failingWarning,
    failingWarningAsync,
    passingWarning,
    passingWarningAsync,
  };
};

export default testDummy;

export const dummyTest = testDummy();
