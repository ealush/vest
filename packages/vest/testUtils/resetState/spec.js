import { getState } from '../../src/core/state';
import { SYMBOL_SUITES } from '../../src/core/state/symbols';
import resetState from '.';

describe('resetState', () => {
  test('Sanity', () => {
    expect(getState()).toMatchSnapshot();
  });

  it('Should create a new state', () => {
    const initialState = getState();
    expect(getState()).toBe(initialState);
    resetState();

    expect(getState()).not.toBe(initialState);
  });

  describe('When invoked with suite name', () => {
    let suiteId;

    beforeEach(() => {
      suiteId = 'test-suite';
    });

    it('Should add a new suite to the state', () => {
      expect(getState(SYMBOL_SUITES)).toEqual({});
      resetState(suiteId);
      expect(getState(SYMBOL_SUITES)[suiteId]).toBeTruthy();
      expect(getState(SYMBOL_SUITES)[suiteId]).toMatchSnapshot();
    });
  });
});
