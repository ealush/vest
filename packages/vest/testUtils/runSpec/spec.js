import runSpec from '.';

describe('module: runSpec', () => {
  let isWatchMode, vestSrc;

  beforeEach(() => {
    isWatchMode = global.isWatchMode;
    vestSrc = require('../../src');
  });

  afterEach(() => {
    global.isWatchMode = isWatchMode;
  });

  describe('jest --watch', () => {
    beforeEach(() => {
      global.isWatchMode = true;
    });

    it('Should run only against source', () => {
      let counter = 0;
      runSpec(vest => {
        counter++;
        expect(vest).toBe(vestSrc);
      });

      expect(counter).toBe(1);
    });
  });

  describe('jest', () => {
    let vestDistVersions;
    beforeEach(() => {
      vestDistVersions = [];
      global.isWatchMode = false;
    });

    it('Should run against source and dist', () => {
      let counter = 0;
      runSpec(vest => {
        counter++;
        vestDistVersions.push(vest);
      });

      expect(vestDistVersions).toContain(vestSrc);
      expect(vestDistVersions).toContain(require('../../dist/vest.js'));
      expect(vestDistVersions).toContain(require('../../dist/vest.min.js'));
      expect(vestDistVersions).toContain(
        require('../../dist/vest.development.js')
      );

      expect(counter).toBe(4);
    });
  });
});
