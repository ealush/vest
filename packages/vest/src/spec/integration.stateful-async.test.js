import resetState from '../../testUtils/resetState';
import runSpec from '../../testUtils/runSpec';
import testDummy from '../../testUtils/testDummy';

const suiteName = 'suite_name';

const suite = ({ create, ...vest }) =>
  create(suiteName, skip => {
    const dummyTest = testDummy(vest);

    vest.skip(skip);
    vest.group('group', () => {
      dummyTest.failingAsync('field_1', 'field_1_group_message');
      dummyTest.failingAsync('field_4', 'field_4_group_message');
    });

    dummyTest.failing('field_1', 'field_statement_1');

    dummyTest.failingAsync('field_2', {
      time: 50,
      statement: 'rejection_message_1',
    });

    dummyTest.passing('field_2', 'field_statement_2');

    dummyTest.passingAsync('field_3', 'field_statement_3');
    dummyTest.failingAsync('field_3', 'field_statement_3');
  });

runSpec(vest => {
  let validate, callback_1, callback_2, callback_3, callback_4, control;

  describe('Stateful async tests', () => {
    beforeEach(() => {
      resetState();
      callback_1 = jest.fn();
      callback_2 = jest.fn();
      callback_3 = jest.fn();
      callback_4 = jest.fn();
      control = jest.fn();
      validate = suite(vest);
    });

    it('Should only run callbacks for last suite run', () =>
      new Promise(done => {
        validate(vest).done(callback_1).done('field_3', callback_2);
        expect(callback_1).not.toHaveBeenCalled();
        expect(callback_2).not.toHaveBeenCalled();
        validate(vest).done(callback_3).done('field_3', callback_4);
        expect(callback_3).not.toHaveBeenCalled();
        expect(callback_4).not.toHaveBeenCalled();
        setTimeout(() => {
          expect(callback_1).not.toHaveBeenCalled();
          expect(callback_2).not.toHaveBeenCalled();
          expect(callback_3).not.toHaveBeenCalled();
          expect(callback_4).toHaveBeenCalled();
          control();
        });
        setTimeout(() => {
          expect(callback_1).not.toHaveBeenCalled();
          expect(callback_2).not.toHaveBeenCalled();
          expect(callback_3).toHaveBeenCalled();
          expect(control).toHaveBeenCalled();
          done();
        }, 50);
      }));

    it('Merges skipped validations from previous suite', () =>
      new Promise(done => {
        const res = validate('group');
        expect(res.hasErrors('field_1')).toBe(true);
        expect(res.tests.field_1.errorCount).toBe(1);
        expect(res.hasErrors('field_2')).toBe(false);
        expect(res.hasErrors('field_3')).toBe(false);
        expect(res.hasErrors('field_4')).toBe(false);
        expect(res).toMatchSnapshot();
        setTimeout(() => {
          const res = vest.get(suiteName);
          expect(res.tests.field_1.errorCount).toBe(1);
          expect(res.hasErrors('field_2')).toBe(false);
          expect(res.hasErrors('field_3')).toBe(true);
          expect(res.hasErrors('field_4')).toBe(false);
          expect(res).toMatchSnapshot();

          validate('field_2').done(res => {
            expect(res.tests.field_1.errorCount).toBe(2);
            expect(res.hasErrors('field_2')).toBe(true);
            expect(res.hasErrors('field_3')).toBe(true);
            expect(res.hasErrors('field_4')).toBe(true);
            expect(res).toMatchSnapshot();
            done();
          });
        });
      }));
  });
});
