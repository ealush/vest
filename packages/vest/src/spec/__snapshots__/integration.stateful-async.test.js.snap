// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stateful async tests Merges skipped validations from previous suite 1`] = `
Object {
  "done": [Function],
  "errorCount": 1,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 5,
  "tests": Object {
    "field_1": Object {
      "errorCount": 1,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 2`] = `
Object {
  "errorCount": 2,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 5,
  "tests": Object {
    "field_1": Object {
      "errorCount": 1,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 1,
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 3`] = `
Object {
  "errorCount": 5,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {
    "group": Object {
      "field_1": Object {
        "errorCount": 1,
        "testCount": 1,
        "warnCount": 0,
      },
      "field_4": Object {
        "errorCount": 1,
        "testCount": 1,
        "warnCount": 0,
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 7,
  "tests": Object {
    "field_1": Object {
      "errorCount": 2,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 1,
      "errors": Array [
        "rejection_message_1",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 1,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_4": Object {
      "errorCount": 1,
      "testCount": 1,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 4`] = `
Object {
  "done": [Function],
  "errorCount": 1,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 5,
  "tests": Object {
    "field_1": Object {
      "errorCount": 1,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 5`] = `
Object {
  "errorCount": 2,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 5,
  "tests": Object {
    "field_1": Object {
      "errorCount": 1,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 1,
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 6`] = `
Object {
  "errorCount": 5,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {
    "group": Object {
      "field_1": Object {
        "errorCount": 1,
        "testCount": 1,
        "warnCount": 0,
      },
      "field_4": Object {
        "errorCount": 1,
        "testCount": 1,
        "warnCount": 0,
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 7,
  "tests": Object {
    "field_1": Object {
      "errorCount": 2,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 1,
      "errors": Array [
        "rejection_message_1",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 1,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_4": Object {
      "errorCount": 1,
      "testCount": 1,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 7`] = `
Object {
  "done": [Function],
  "errorCount": 1,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 5,
  "tests": Object {
    "field_1": Object {
      "errorCount": 1,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 8`] = `
Object {
  "errorCount": 2,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 5,
  "tests": Object {
    "field_1": Object {
      "errorCount": 1,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 1,
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 9`] = `
Object {
  "errorCount": 5,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {
    "group": Object {
      "field_1": Object {
        "errorCount": 1,
        "testCount": 1,
        "warnCount": 0,
      },
      "field_4": Object {
        "errorCount": 1,
        "testCount": 1,
        "warnCount": 0,
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 7,
  "tests": Object {
    "field_1": Object {
      "errorCount": 2,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 1,
      "errors": Array [
        "rejection_message_1",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 1,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_4": Object {
      "errorCount": 1,
      "testCount": 1,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 10`] = `
Object {
  "done": [Function],
  "errorCount": 1,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 5,
  "tests": Object {
    "field_1": Object {
      "errorCount": 1,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 11`] = `
Object {
  "errorCount": 2,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 5,
  "tests": Object {
    "field_1": Object {
      "errorCount": 1,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 1,
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`Stateful async tests Merges skipped validations from previous suite 12`] = `
Object {
  "errorCount": 5,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {
    "group": Object {
      "field_1": Object {
        "errorCount": 1,
        "testCount": 1,
        "warnCount": 0,
      },
      "field_4": Object {
        "errorCount": 1,
        "testCount": 1,
        "warnCount": 0,
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_name",
  "testCount": 7,
  "tests": Object {
    "field_1": Object {
      "errorCount": 2,
      "errors": Array [
        "field_statement_1",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "field_2": Object {
      "errorCount": 1,
      "errors": Array [
        "rejection_message_1",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "field_3": Object {
      "errorCount": 1,
      "testCount": 2,
      "warnCount": 0,
    },
    "field_4": Object {
      "errorCount": 1,
      "testCount": 1,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;
